### PR TITLE
Some lovely jubly hotfixes

### PR DESCRIPTION
**Pull request by binaryoverload**
- [x] Completed PR - Has all features or fixes that it was meant to and isn't half done
- [x] Follows the style guide - Style guide found [here](https://github.com/FlareBot/FlareBot/blob/master/CONTRIBUTING.md)
- [x] All changes have been tested

## Added
 - GitHandler to manage git

## Changed
 - Sentry settings are now *all* set programmatically
 - Sentry release is now set to the git commit hash not the version #

## Fixed
 - Not taking into account channel overrides for perms
 - Audit log optimisation

### CODE REVIEW

**Comment by WalshyDev** [comment]
Also you didn't modify the build.gradle also - I want to maybe phase out Maven and rely more on Gradle (since we kinda need to for err *nudge nudge*). Plus Travis seems to prioritise Gradle even when it's told to build with Maven...

**Comment by binaryoverload** [comment]
Oh shit xDD Yeah I'll do that now

File: src/main/java/stream/flarebot/flarebot/ModlogEvents.java [hunk 1]
```diff
@@ -54,11 +54,13 @@
     @Override
     public void onGuildBan(GuildBanEvent event) {
         if (cannotHandle(event.getGuild(), ModlogEvent.USER_BANNED)) return;
-        event.getGuild().getAuditLogs().limit(1).queue(auditLogEntries -> {
+        event.getGuild().getAuditLogs().limit(1).type(ActionType.BAN).queue(auditLogEntries -> {
             AuditLogEntry entry = auditLogEntries.get(0);
+            boolean validEntry = true;
+            validEntry = entry.getTargetId().equals(event.getUser().getId());
```
[the remark this answers is not in the document]
Comment (binaryoverload) [reply src/main/java/stream/flarebot/flarebot/ModlogEvents.java]: Long dayyyy

**Comment by weeryan17** [comment]
@binaryoverload while your hear fix unmuting. I believe it's still broken

File: src/main/java/stream/flarebot/flarebot/GitHandler.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import java.io.IOException;
+import java.util.Map;
+
+public class GitHandler {
+
+    private static final Git GIT;
+    private static RevCommit LATEST_COMMIT;
+
+    static {
+
```
Comment (WalshyDev) [inline src/main/java/stream/flarebot/flarebot/GitHandler.java]: Remove this space 👀

File: src/main/java/stream/flarebot/flarebot/GitHandler.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import org.eclipse.jgit.revwalk.RevWalk;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Map;
+
+public class GitHandler {
+
+    private static final Git GIT;
+    private static RevCommit LATEST_COMMIT;
```
Comment (WalshyDev) [inline src/main/java/stream/flarebot/flarebot/GitHandler.java]: This isn't a constant anymore therefore shouldn't be fully capitalised

**Comment by binaryoverload** [comment]
@WalshyDev could we possibly get this merged for the update today? I can't
do anymore updates tho :eyes:

**Pull request merged**